datefinder.py

```diff
@@ -156,6 +159,7 @@ def _find_and_replace(self, date_string, captures):
         for key, replacement in self.REPLACEMENTS.items():
             date_string = re.sub(key,replacement,date_string,flags=re.IGNORECASE)
 
+        self.REPLACEMENTS = original_replacements
```

Comment:
If this is the only function we use ``self.REPLACEMENTS`` in, why don't we just mutate the copy?